File: fizzbuzz_program.py
## TheJollyDuck
## FizzBuzz
## This program prints out numbers but for
## a specifc multiple of a number with a word

numList = [3,5,10,13]
wordlist = ["fizz","buzz","fizzle","kaboom"]

print("FizzBuzz is starting....")

# TO DO : Implement input safety so that it only accepts ints
numberLimit = int(input("Up to what number do you want to print out? [int]: "))

for i in range(0, numberLimit):         # This for loop iterates until the user-defined limit is reached
    output = ""                         # contains the final string for output
    for j in range(0,len(numList)):     # iterates all those through a list
        if i%numList[j] == 0:           # Checks if number is a multiple of defined number
            output += wordlist[j]       # If passes, concatenates the string
    if output == "":                    # Checks if output is empty
        print(i)                        # If passes, prints out number
    else:
        print(output)                   # Otherwise, print out string

End = input("Press [Enter] to quit application....")

File: fizzbuzz_program.c
/*
 * TheJollyDuck
 *
 * Fizz Buzz Program
 * This is a program that generates
 * the patterns necessary for the game
 * Fizz Buzz
 * Has some limitations regardin the output size,
 * without dynamic arrays.
 *
 * Originally had a custom function handling
 * concatenation but had problems with it,
 * and I also found <string.h>
 * Once I found it , it was smooth sailing.
 *
 */

#include <stdio.h>                                              // For i/o
#include <stdlib.h>                                             // Standard library
#include <string.h>                                             // for string functions

typedef struct pair {                                           // Struct for easy implementation of a word and it's number

    char word[20];
    int number;

} pattern;

const int ARRAY_SIZE = 20;                                      // Array size must be bigger than total pattern chars
void printOutput(char a[], int b);                              // To decide on whether to output number or pattern

int main(void){
    pattern req[] = {{"Fizz", 3}, {"Buzz", 5}, {"Jazz", 8}};    // Contains the word and associated number
    int numberLimit;

    printf("Up to what number do you want to print out? [int]: ");
    scanf("%d", &numberLimit);                                  // User input to see how much

    for (int i = 1; i < numberLimit; i++){                      // Iterates through each number up to user-defined limit
        char output[ARRAY_SIZE];
        output[0] = '\0';                                       // Declares and initializes an empty string for concatenation

        for (int j = 0; j < (sizeof(req)/sizeof(req[0])); j++) {// Iterates through each pattern to check for multiples
            if (i % req[j].number == 0) {
                strcat(output, req[j].word);                    // If it is a multiple, the name is concatenated to
            }
        }
        printOutput(output, i);                                 // Prints the coressponding output for the number
    }

    printf("Press [Enter] to end program");
    getchar();
    return 0;
}

/*
 * printOutput()
 * This function is to check whether we have
 * an integer or a string to output.
 *
 * This method forgoes any need for casting
 * and was easier to implement.
 *
 * Originally tried to do it the python way
 * but failed.
 *
 */

void printOutput(char a[], int b){                              // Params are the output array & the number
    if (a[0] == '\0') {
        printf("%d\n",b);                                       // If output string is empty, print number

    } else {
        printf("%s\n", a);                                      // Otherwise, print the output string

    }
}
